pyfr/backends/openmp/kernels/batch-gemm.mako

```diff
@@ -4,19 +4,38 @@
 struct kargs
 {
     void (*exec)(void *, const fpdtype_t *, fpdtype_t *);
-    void *blockk;
     int nblocks;
     const fpdtype_t *b;
     int bblocksz;
     fpdtype_t *c;
     int cblocksz;
+    void *blockk[];
 };
 
 void batch_gemm(const struct kargs *restrict args)
 {
     #pragma omp parallel for ${schedule}
     for (int ib = 0; ib < args->nblocks; ib++)
-        args->exec(args->blockk,
+    % if nfac == 1:
+        args->exec(args->blockk[0],
                    args->b + ib*args->bblocksz,
                    args->c + ib*args->cblocksz);
+    % else:
+    {
+        fpdtype_t buf1[cblocksz], buf2[cblocksz];
+      % for i in range(nfac):
```

Comment:
This code might be simpler if you reversed the order of items in blockk.  Also inline expressions might help ${1 + i % 2} or something to that effect for picking buffer numbers.